lib/stm32/g0/flash.c

```diff
@@ -30,6 +30,145 @@
 
 #include <libopencm3/stm32/flash.h>
 
+/** @brief Wait until Last Flash Operation has Ended */
+void flash_wait_for_last_operation(void)
+{
+	while ((FLASH_SR & FLASH_SR_BSY) == FLASH_SR_BSY);
+}
+
+/** @brief Program a 64bits word to FLASH
+ *
+ * Program a 64bit word to FLASH memory.
+ * Flash programing error must be checked and cleared if needed
+ * before starting programming.
+ *
+ * @param[in] address Address in FLASH
+ * @param[in] data Double word to write
+ */
+void flash_program_double_word(uint32_t address, uint64_t data)
+{
+	flash_wait_for_last_operation();
+
+	FLASH_CR |= FLASH_CR_PG;
+
+	MMIO32(address) = (uint32_t)data;
+	MMIO32(address+4) = (uint32_t)(data >> 32);
+
+	flash_wait_for_last_operation();
+
+	FLASH_CR &= ~FLASH_CR_PG;
+}
+
+/** @brief Program a Data Block to FLASH
+ *
+ * This programs an arbitrary length data block to FLASH memory.
+ * The program error flag should be checked separately for the event that memory
+ * was not properly erased.
+ *
```

Comment:
might be worth mentioning that it is min 8 bytes here?  (i guess that's common fialing of the docs really.)

Reply:
Indeed, same issue for other chips, that code ends up writting end of data garbage if not properly sized.
I wondered about putting an assert on data size ? might be a bit too much brutal

Reply:
yeah, we can just leave it.  not in scope for this at least.